web/main/models.py

```diff
@@ -1429,6 +1429,29 @@ class Meta:
         unique_together = (("user", "casebook"),)
 
 
+class CasebookTag(models.Model):
```

Comment:
Usually the reason to use a through model rather than just a direct many-to-many is to add additional fields on the relationship. However, it's so annoying to change your mind later that I usually recommend starting with a through model—so I think this is right. Did you have additional fields in mind you might want to add here? I could imagine, for example, recording the user who did the tagging operation.

Reply:
Yes, that's what I was thinking. I didn't have anything specific in mind, more just future-proofing. I think I also just like being able to see the table for myself, in a way.

User makes sense; I'll add that in! The other thing I was thinking of was `type` if we are going to link to more than just `Casebook`s.

Reply:
Having looked at the plans for this, it seems like it would be helpful to have a type for the tag itself, rather than just the object it points to. E.g. topic and institution tags both point to casebooks, but they represent something different, so maybe se should note that --- probably on the `CasebookTag` model (the old `CasebookTagName`)